processor_function/shared/cnab_builder.py

```diff
@@ -29,22 +29,41 @@ def __init__(self, resource_request_message: dict):
         self._container_group_name = ""
         self._location = ""
 
-    def _build_porter_cmd_line(self) -> str:
+    def _build_porter_cmd_line(self) -> List[str]:
         porter_parameters = ""
         for key in self._message['parameters']:
             porter_parameters += " --param " + key + "=" + self._message['parameters'][key]
 
-        start_command_line = "/bin/bash -c porter " + self._message['operation'] + " CNAB --tag " + self._message[
-            'bundle-name'] + porter_parameters + " -d azure && porter show CNAB"
-
-        logging.info("Creating a runner with:" + start_command_line)
+        for key, value in os.environ.items():
+            if key.startswith("param_"):
+                porter_parameters += " --param " + key.replace('param_', '') + "=" + value
+
+        installation_id = self._message['parameters']['tre_id'] + "-" + self._message['parameters']['workspace_id']
+        start_command_line = ["/bin/bash", "-c", "az login --identity "
+                              + "&& TOKEN=$(az acr login --name msfttreacr --expose-token --output tsv --query accessToken) "
```

Comment:
Yeah, this is the way! Concatenating strings on multiple lines the plus sign comes in the beginning. Also liking the "&&" in the beginning. Very nice (except for the full indentation which I don't like :( )!

Reply:
changed indentation